### ISSUE TITLE
panic: reflect: NumField of non-struct type

### ISSUE THREAD

**Issue by Kvaz1r** [timshannon/bolthold#103]
Today I've updated the library and after recompilation my application got such panic:
`panic: reflect: NumField of non-struct type`

for line:

>... /Gopath/src/github.com/timshannon/bolthold/get.go:62 +0x274

It was introduced in this commit - [Started on adding key-handling for Get](https://github.com/timshannon/bolthold/commit/4f121e80e817e0d784b1899582cf9465d4412bd4)

MCVE:

```
package main

import (
	"log"

	"github.com/timshannon/bolthold"
)

type Counterer struct {
	State uint
}

var count = new(Counterer)

func main() {
	store, err := bolthold.Open("store", 0666, nil)
	checkErr(err)
	count.State++
	checkErr(store.Upsert("count", count))
	err = store.Get("count", &count)
	log.Println(err)
}

func checkErr(err error) {
	if err != nil {
		log.Fatalln(err)
	}
}
```

**Comment by timshannon** [comment]
This is fixed now in master with #104.

I made changed to better handle keys like `Find` queries. However technically you aren't `Getting` the same type as you are `Upserting`.

You upsert a `*Counterer`, but you are trying to get it into a `**Counterer`.  By passing `&count` into your `store.Get` call you are getting a pointer to your pointer.

Regardless, I updated to code so it handles this without panicing, but I thought I'd let you know.

Thanks,
